enhancements/microshift/kubernetes-for-industrial-edge.md

```diff
@@ -0,0 +1,435 @@
+UltraScale+, fitlet2, NVIDIA Jetson TX2, or Intel NUCs that are
+pervasive in edge deployments. These cost a few hundred USD, have all
+necessary accelerators and I/O on-board, but are typically not
+extensible and are highly constrained on every resource, e.g.:
+
+* CPU: ARM Cortex or Intel Atom class CPU, 2-4 cores, 1.5GHz clock
+* memory: 2-16GB RAM
+* storage: e.g. SATA3 @ 6Gb/s, 10kIOPS, 1ms (vs. NVMe @ 32Gb/s,
+  500kIOPS, 10µs)
+* network: Less than data center-based servers. Likely 1Gb/s NICs,
+  potentially even LTE/4G connections at 5-10Mbps, instead of
+  10/25/40Gb/s NICs
+
+These use cases are not being addressed by RHEL or OpenShift
+today. RHEL doesn’t include Kubernetes. OpenShift is fundamentally
+designed for a cloud-centric and server-centric world, with
+operational models and assumptions about environmental constraints
+that are very different from field-deployed devices. Even single-node
+OpenShift deployments target traditional multi-socket data center
+computing platforms, not the small edge devices targeted by
+MicroShift.
+
+### User Stories
+
+An application developer builds applications.
+
+An application deployer packages and distributes those applications.
+
+A device administrator manages the lifecycle of a device running the
+applications.
+
+1. As an application developer, I can design my application deployment
+   so that it works with MicroShift and all other topologies of
+   OpenShift in a consistent way.
+2. As an application deployer, I can package MicroShift, the
+   applications, and any other platform components into an image that
+   is pre-installed on many devices by the manufacturer.
+3. As an application deployer, I can package MicroShift, the
+   applications, and any other platform components into an image that
+   I install myself.
+4. As an device administrator, I can release an updated image over the
+   network to upgrade devices already in the field using minimal
+   network bandwidth.
+5. As an device administrator, I can rely on the system to rollback to
+   a good state if an upgrade fails.
+6. As an device administrator, I have control over the operating
+   system running under MicroShift.
+
+### Goals
+
+* Describe an approach to providing a version of Kubernetes compatible
+  with the OpenShift APIs that it includes and that meets the
+  constraints described in the Motivation section.
+* Identify the minimal set of APIs that will be included and the
+  components needed to support them.
+* Design a system that can work autonomously, without requiring
+  external orchestration tools for deployment.
+* Treat MicroShift as an application running on RHEL, with minimal
+  privileges.
+* Design a system that can be configured by someone familiar with
+  Linux, without Kubernetes expertise.
+* Follow the [MicroShift design
+  principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* We are not trying to fit all of OpenShift into the target hardware
```

Comment:
The term OpenShift is overloaded in this context when describing a new form-factor.

Just like HyperShift is a project name, but the product usage is "hosted control planes".

MicroShift is a project name, not a product name.  In this context, contrast with "standalone OpenShift deployments".

Reply:
Updated